File: Data Structures & Algorithms/validate-parentheses/submission-2.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> brackets;
        for (char i : s){
            if (brackets.empty() && (i == ')' || i == '}' || i == ']')){
                brackets.push(i);
            }
            else{
            if (i == '(' || i == '{' || i == '['){
                brackets.push(i);
            }
            else{
                if ((i == ')' && brackets.top()=='(') || (i == '}' && brackets.top()=='{') || (i == ']' && brackets.top()=='[')){
                    brackets.pop();
                }
                else{
                    brackets.push(i);
                }
            }
            }
        }
        if (brackets.empty()){
            return true;
        }
        else{
            return false;
        }
    }
};

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-2.py
import operator
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        operators = {
            "+": operator.add,
            "-": operator.sub,
            "*": operator.mul,
            "/": operator.truediv
        }
        stack = []
        for t in tokens:
            if t in operators:
                #print(f"t is an operator: {t}")
                second = int(stack.pop())
                #print(f"second is: {second}")
                first = int(stack.pop())
                #print(f"first is: {first}")
                #print(f"operator is: {t}")
                calcul = operators[t](first, second)
                #print(f"calcul is: {calcul}")
                stack.append(calcul)
            else:
                stack.append(t)
                #print (f"This is the current stack: {stack}") 
                
        return int(stack.pop())